### PR TITLE
HPCC-23772 PKI Fix for _ in compnames and default public keys

### PR DESCRIPTION
**Pull request by Michael-Gardner**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [x] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@mayx Please take a look at this.  Fixes the _ issue on some environments where names like thor_after are getting stripped apart and the setupPKI script is outputting keys for 'after' components instead of 'thor_after'.  Also includes default keys and ensures everything is properly owned at the end of the process by the user.

File: initfiles/bash/etc/init.d/setupPKI.in [hunk 1]
```diff
@@ -73,7 +73,7 @@ fi
 for i in "" ${compArray[@]} ; do
     compName=""
     if [ "$i" != "" ]; then
-        compName=""${i#*_}
```
Comment (mayx) [inline initfiles/bash/etc/init.d/setupPKI.in:76]: I remember vaguely that you added "#*_" to deal with an issue in 160, would removing it break that fix?

File: initfiles/bash/etc/init.d/setupPKI.in [hunk 2]
```diff
@@ -124,4 +124,6 @@ else
     log_success_msg
 fi
 
+chown -R ${USER_NAME}:${USER_NAME} ${homePath}
```
Comment (mayx) [inline initfiles/bash/etc/init.d/setupPKI.in:127]: Did you mean ${certPath}? ${homePath} is the home directory of the user, don't think it should be touched by this script...

**Review by mayx — COMMENTED** [review]
@Michael-Gardner a couple of comments...

File: initfiles/bash/etc/init.d/setupPKI.in (hunk 1 above)
Comment (Michael-Gardner) [reply initfiles/bash/etc/init.d/setupPKI.in:76]: This I think was originally from your commit a149d5694c917a4cef114eb345919c15284d165c.  I think we talked about possibly changing it but I don't think we ever did.

File: initfiles/bash/etc/init.d/setupPKI.in (hunk 2 above)
Comment (Michael-Gardner) [reply initfiles/bash/etc/init.d/setupPKI.in:127]: yeah I think you're right, it should be touching the /home/hpcc/certificate folder (or whatever {certPath} is set to and not everything under homePath

**Comment by Michael-Gardner** [comment]
@mayx Fixed the homePath to certPath.  Everything else looks good here in my testing. Can you take one last look?

**Review by mayx — APPROVED** [review]
@Michael-Gardner looks good.

**Comment by Michael-Gardner** [comment]
@richardkchapman ready for merge

**Pull request merged**